iml-agent/src/lustre.rs

```diff
@@ -4,23 +4,38 @@
 
 use crate::agent_error::ImlAgentError;
 use futures::TryFutureExt;
-use iml_cmd::{CheckedCommandExt, Command};
+use iml_cmd::{CheckedCommandExt, CmdError, Command};
 use liblustreapi::LlapiFid;
 use std::ffi::OsStr;
+use std::time::Duration;
 use tokio::task::spawn_blocking;
+use tokio::time::delay_for;
 
 /// Runs lctl with given arguments
 pub async fn lctl<I, S>(args: I) -> Result<String, ImlAgentError>
 where
-    I: IntoIterator<Item = S>,
+    I: IntoIterator<Item = S> + Clone,
     S: AsRef<OsStr>,
 {
-    Command::new("/usr/sbin/lctl")
-        .args(args)
-        .checked_output()
-        .err_into()
-        .await
-        .map(|o| String::from_utf8_lossy(&o.stdout).to_string())
+    loop {
+        let r = Command::new("/usr/sbin/lctl")
+            .args(args.clone())
+            .checked_output()
+            .await;
+
+        if let Err(CmdError::Output(o)) = &r {
```

Comment:
It seems like this should wrap the `lctl` call as a new fn instead of being within it.

Reply:
I did it first, but the function was private anyway.

Reply:
Done

Reply:
Maybe call this fn `lctl_retry`, and keep the original `lctl` fn in place. the `lctl` fn is being used in a few other modules already, and it's behavior should stay the same I think.

Reply:
Actually, other modules will just become more sound out of the box. I don't think there is sense in failing for a temporarily unavailable resource.

Reply:
Have we seen this on any other lctl call (and do we know the behavior is the same)?

Reply:
Ah, that's valid question. As a precaution we indeed can have two version of lctl function. But the lctl executable is using perror() or strerror() consistently. And retrying temporary failures is still needed.

It is really simple: we would loop iff lctl prints "Resource temporarily unavailable" (to stderr + exit code, it could not be a snapshot comment :) )

Reply:
Maybe add the call into IML's liblustreapi library?  Might be cleaner than doing a shell command.

Reply:
Make it `lctl_retry`